### PR TITLE
Switch to pyproject.toml

### LINKED ISSUE #813: TODO: Switch to pyproject.toml

### PR DESCRIPTION
**Pull request by raman325**
fixes #813

### CODE REVIEW

**Comment by raman325** [comment]
One thing I noticed while testing, here is the SOURCES.txt file in the egg-info folder, why are the `test/*.py` files all being included?

```
LICENSE
README.md
pyproject.toml
setup.py
test/test_client.py
test/test_dump.py
test/test_event.py
test/test_firmware.py
test/test_main.py
test/test_version.py
zwave_js_server/__init__.py
zwave_js_server/__main__.py
zwave_js_server/client.py
zwave_js_server/dump.py
zwave_js_server/event.py
zwave_js_server/exceptions.py
zwave_js_server/firmware.py
zwave_js_server/py.typed
zwave_js_server/version.py
zwave_js_server/const/__init__.py
zwave_js_server/const/command_class/__init__.py
zwave_js_server/const/command_class/barrier_operator.py
zwave_js_server/const/command_class/basic.py
zwave_js_server/const/command_class/central_scene.py
zwave_js_server/const/command_class/color_switch.py
zwave_js_server/const/command_class/energy_production.py
zwave_js_server/const/command_class/entry_control.py
zwave_js_server/const/command_class/humidity_control.py
zwave_js_server/const/command_class/lock.py
zwave_js_server/const/command_class/meter.py
zwave_js_server/const/command_class/multilevel_sensor.py
zwave_js_server/const/command_class/multilevel_switch.py
zwave_js_server/const/command_class/notification.py
zwave_js_server/const/command_class/power_level.py
zwave_js_server/const/command_class/protection.py
zwave_js_server/const/command_class/scene_activation.py
zwave_js_server/const/command_class/sound_switch.py
zwave_js_server/const/command_class/thermostat.py
zwave_js_server/const/command_class/wake_up.py
zwave_js_server/const/command_class/window_covering.py
zwave_js_server/model/__init__.py
zwave_js_server/model/association.py
zwave_js_server/model/command_class.py
zwave_js_server/model/device_class.py
zwave_js_server/model/device_config.py
zwave_js_server/model/driver.py
zwave_js_server/model/duration.py
zwave_js_server/model/endpoint.py
zwave_js_server/model/log_config.py
zwave_js_server/model/log_message.py
zwave_js_server/model/notification.py
zwave_js_server/model/statistics.py
zwave_js_server/model/utils.py
zwave_js_server/model/value.py
zwave_js_server/model/version.py
zwave_js_server/model/controller/__init__.py
zwave_js_server/model/controller/data_model.py
zwave_js_server/model/controller/event_model.py
zwave_js_server/model/controller/firmware.py
zwave_js_server/model/controller/inclusion_and_provisioning.py
zwave_js_server/model/controller/rebuild_routes.py
zwave_js_server/model/controller/statistics.py
zwave_js_server/model/node/__init__.py
zwave_js_server/model/node/data_model.py
zwave_js_server/model/node/event_model.py
zwave_js_server/model/node/firmware.py
zwave_js_server/model/node/health_check.py
zwave_js_server/model/node/statistics.py
zwave_js_server/util/__init__.py
zwave_js_server/util/helpers.py
zwave_js_server/util/lock.py
zwave_js_server/util/multicast.py
zwave_js_server/util/node.py
zwave_js_server/util/command_class/__init__.py
zwave_js_server/util/command_class/energy_production.py
zwave_js_server/util/command_class/meter.py
zwave_js_server/util/command_class/multilevel_sensor.py
zwave_js_server_python.egg-info/PKG-INFO
zwave_js_server_python.egg-info/SOURCES.txt
zwave_js_server_python.egg-info/dependency_links.txt
zwave_js_server_python.egg-info/entry_points.txt
zwave_js_server_python.egg-info/requires.txt
zwave_js_server_python.egg-info/top_level.txt
```

File: setup.py [hunk 1]
```diff
@@ -1,38 +1,4 @@
 """Setup module for zwave-js-server-python."""
```
Comment (MartinHjelmare) [inline setup.py:1]: Do we still need setup.py?

**Review by MartinHjelmare — APPROVED** [review]

**Comment by MartinHjelmare** [comment]
> why are the test/*.py files all being included?

Good question. I don't know.

File: setup.py (hunk 1 above)
Comment (raman325) [reply setup.py]: thought we did but everything seemed to work as expected without it, thanks!

**Comment by raman325** [comment]
> > why are the test/*.py files all being included?
> 
> Good question. I don't know.

answered here: https://packaging.python.org/en/latest/guides/using-manifest-in/#how-files-are-included-in-an-sdist

**Comment by raman325** [comment]
I can use `MANIFEST.in` to get rid of the test files. Should I? I'm not sure what value they add to the distribution

**Comment by raman325** [comment]
> Maybe just rename the directory to `tests`?

Same problem

**Comment by MartinHjelmare** [comment]
Ok for me to add a MANIFEST.in file.

**Pull request merged**